### DISCUSSION TITLE
couch returned 400

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#9016] [Help]
Hi,

Error when navigating screen in published app and table design access.

Table page not showing data rows.  Table has data before publishing app and new rows added in published app.

Budibase Cloud
App Version : 2.2.1

published app error on navigated to screen
![image](https://user-images.githubusercontent.com/42922833/207116729-fc9f3d0d-b62f-4307-ae46-8efeece61e8d.png)

![image](https://user-images.githubusercontent.com/42922833/207116775-b2f4641e-fa1a-4cda-83be-ab5b72e356f6.png)

Error when table accessed
![image](https://user-images.githubusercontent.com/42922833/207118336-9797cdda-92e1-4ba1-85f1-ad8cf872ce81.png)

**Comment by andz-bb** [comment]
hey @applicationuser, which network request is giving you that response? thanks

**Reply by applicationuser** [reply]
Hi @andz-bb 

looks like api request to fetch table rows  

![image](https://user-images.githubusercontent.com/42922833/207392631-f5dda796-f62a-4c16-966a-c9f86c6bad93.png)

now automatons also failing in published app - this is for another app which has cron based triggered that failed 

  "response": "Error: couch returned 400"
   
![image](https://user-images.githubusercontent.com/42922833/207393367-6cc4a457-b9b2-4794-9307-a0eb4f5160fb.png)

**Comment by andyburgessmd** [comment]
I am also seeing this today.
Console shows this
![image](https://user-images.githubusercontent.com/110813989/207312502-5543cdc4-b324-4b76-94b3-06a9b99c0a75.png)

**Reply by andyburgessmd** [reply]
This is from the application not builder. 
My Users table in the builder is also showing blank

**Comment by mike12345567** [comment]
Hi guys - can anyone provide an export of an app exhibiting this behaviour? I've not been able to re-create this so far.

**Comment by shogunpurple** [comment]
Hi @applicationuser - I imported your app and it seems to work for me, I'm having trouble reproducing. Have you tried republishing your app? Does the issue persist?

**Reply by stellarfinn** [reply]
I have three versions of the app I'm working on... they're all displaying this same error. It's a Budibase issue, not user. Loss of table data in one of the database tables and the error "couch returned 400."

**Reply by shogunpurple** [reply]
@stellarfinn I’m aware - I’m not saying it’s a user issue. I was just wondering if a publish might have resolved it for them

**Comment by stellarfinn** [comment]
I'm getting the same error. Haven't logged in to Budibase to work on the app a couple weeks so nothing has changed as far as building the app out goes. Seems to be a Budibase issue?

**Reply by stellarfinn** [reply]
I've looked at my database in the builder and they're empty and the "couch returned 400" error shows up there too. 



![Screenshot 2022-12-13 141400](https://user-images.githubusercontent.com/99847728/207444868-891d957c-9967-434c-a6a6-d17d6e502b01.jpg)

**Reply by stellarfinn** [reply]
I'm assuming the Budibase cloud database is couch or a couch fork? I'm so happy I'm not paying for this, my boss just asked for a review of the app I'm developing for our business and of course I login to show him and this happens.

**Reply by shogunpurple** [reply]
@stellarfinn what is your tenantId? We are investigating but having a really hard time reproducing this on our side, we will update the thread once we find the issue and push a fix.

**Reply by stellarfinn** [reply]
Where do I find the tenantId?

**Reply by shogunpurple** [reply]
It’s in the URL  - yourtenant.budibase.app

**Reply by stellarfinn** [reply]
https://ciaoimports.budibase.app/app/portal

**Reply by stellarfinn** [reply]
The bizarre bit is nothing has changed on my end. It was working fine two weeks or so ago when I reviewed it. Hadn't updated to the latest version of Budibase or anything. That is problematic, I'm advocating for paying for Budibase to my superior and this just made it more difficult. I love Budibase and the time it saves but it can't just manifest problems out of nowhere so regularly if you're paying for a premium subscription.

**Reply by shogunpurple** [reply]
We have a fix and it will be rolled out ASAP - thanks for the debugging info

**Comment by stellarfinn** [comment]
Thank you. Can you elaborate on the cause?

On Tue, Dec 13, 2022 at 4:50 PM Martin McKeaveney ***@***.***>
wrote:

> We have a fix and it will be rolled out ASAP - thanks for the debugging
> info
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/9016#discussioncomment-4395043>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXZY4METJFIV52K7TAMLLBDWNEDURANCNFSM6AAAAAAS4HR2RI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Sincerely,

*Bobby Segurola*
*(954) 736-7836*
*--*

P Please consider the environment before printing this email.

CONFIDENTIALITY NOTICE: This e-mail transmission, and any documents,
files or previous e-mail messages attached to it, may contain confidential
information that is legally privileged. If you are not the intended
recipient, or a person responsible for delivering it to the intended
recipient, you are hereby notified that any disclosure, copying,
distribution or use of any of the information contained in or attached
to this message is STRICTLY PROHIBITED.

**Reply by shogunpurple** [reply]
The issue was down to essentally load - we recently moved to a new CouchDB client which queries the database in a slightly different way, querying each ID by passing a series of query parameters - when you have a lot of relationships on a record, the URL got very long and this caused the error. We've forked the client library and made an update to prevent this behaviour, which will fix it. This issue only affects those with quite a lot of data in a relationship (could be created by, updated by records by a certain user for example)

**Reply by mike12345567** [reply]
Hi @stellarfinn - the PR for the fix is here: https://github.com/Budibase/budibase/pull/9038

As part of some recent refactoring work in Budibase we swapped out our interface for CouchDB from the library `pouchdb` to a lighter, simple HTTP interface called `couchdb-nano`. The Nano library is maintained by the CouchDB maintainer team, but changes one core thing about how we interfaced with CouchDB, essentially it changed our request for documents from a `POST` request, with a body containing the query parameters, to a `GET` request with parameters in the query string instead. It turned out in some scenarios utilising the query string to send parameters caused CouchDB to respond with various errors, vs using the post request method and the body for parameters being more full proof. To solve this we've forked the library and made some updates to it - we should be able to improve the library a bit for use with Budibase as well this way.

**Comment by stellarfinn** [comment]
Awesome. Thanks for the breakdown.

On Tue, Dec 13, 2022 at 5:10 PM Michael Drury ***@***.***>
wrote:

> Hi @stellarfinn <https://github.com/stellarfinn> - the PR for the fix is
> here: #9038 <https://github.com/Budibase/budibase/pull/9038>
>
> As part of some recent refactoring work in Budibase we swapped out our
> interface for CouchDB from the library pouchdb to a lighter, simple HTTP
> interface called couchdb-nano. The Nano library is maintained by the
> CouchDB maintainer team, but changes one core thing about how we interfaced
> with CouchDB, essentially it changed our request for documents from a POST
> request, with a body containing the query parameters, to a GET request
> with parameters in the query string instead. It turned out in some
> scenarios utilising the query string to send parameters caused CouchDB to
> respond with various errors, vs using the post request method and the body
> for parameters being more full proof. To solve this we've forked the
> library and made some updates to it - we should be able to improve the
> library a bit for use with Budibase as well this way.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Budibase/budibase/discussions/9016#discussioncomment-4395200>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXZY4MDQ2BOXXELTYD247NTWNEF6ZANCNFSM6AAAAAAS4HR2RI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Sincerely,

*Bobby Segurola*
*(954) 736-7836*
*--*

P Please consider the environment before printing this email.

CONFIDENTIALITY NOTICE: This e-mail transmission, and any documents,
files or previous e-mail messages attached to it, may contain confidential
information that is legally privileged. If you are not the intended
recipient, or a person responsible for delivering it to the intended
recipient, you are hereby notified that any disclosure, copying,
distribution or use of any of the information contained in or attached
to this message is STRICTLY PROHIBITED.

**Comment by shogunpurple** [comment]
Hi folks - the latest change has now been deployed to budibase cloud and self host. We will close the issue out here if everyone is back up and running. Apologies for the service interruption.
